encoder_test.go

```diff
@@ -129,6 +129,71 @@ func uint64TestCases() []compactTestCase[uint64] {
 	}
 }
 
+func newInteger[T ~uint8 | ~uint16 | ~uint32 | ~uint64](v T) *T {
```

Comment:
NIT: name could be more explanatory:

```suggestion
func intPtr[T ~uint8 | ~uint16 | ~uint32 | ~uint64](v T) *T {
```